pkg/replicaset/replicaset.go

```diff
@@ -0,0 +1,469 @@
+		glog.V(100).Infof("The 'nodeSelector' of the replicaset is empty")
+
+		builder.errorMsg = "can not apply empty nodeSelector"
+
+		return builder
+	}
+
+	for key := range nodeSelector {
+		if key == "" {
+			glog.V(100).Infof("The 'nodeSelector' key value cannot be empty")
+
+			builder.errorMsg = "can not apply a nodeSelector with an empty key value"
+
+			return builder
+		}
+	}
+
+	builder.Definition.Spec.Template.Spec.NodeSelector = nodeSelector
+
+	return builder
+}
+
+// WithVolume defines Volume of replicaset under ContainerTemplateSpec.
+func (builder *Builder) WithVolume(rsVolume corev1.Volume) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	if rsVolume.Name == "" {
+		glog.V(100).Infof("The Volume name parameter is empty")
+
+		builder.errorMsg = "volume name parameter is empty"
+
+		return builder
+	}
+
+	glog.V(100).Infof("Adding volume %s for replicaset %s container template in namespace %s",
+		rsVolume.Name, builder.Definition.Name, builder.Definition.Namespace)
+
+	builder.Definition.Spec.Template.Spec.Volumes = append(
+		builder.Definition.Spec.Template.Spec.Volumes,
+		rsVolume)
+
+	return builder
+}
+
+// WithAdditionalContainerSpecs appends a list of container specs to the replicaset definition.
+func (builder *Builder) WithAdditionalContainerSpecs(specs []corev1.Container) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Appending a list of container specs %v to replicaset %s in namespace %s",
+		specs, builder.Definition.Name, builder.Definition.Namespace)
+
+	if len(specs) == 0 {
+		glog.V(100).Infof("The container specs are empty")
+
+		builder.errorMsg = "cannot accept empty list as container specs"
+
+		return builder
+	}
+
+	builder.Definition.Spec.Template.Spec.Containers = append(builder.Definition.Spec.Template.Spec.Containers, specs...)
+
+	return builder
+}
+
+// Exists checks whether the given replicaset exists.
+func (builder *Builder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if replicaset %s exists in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+	builder.Object, err = builder.apiClient.ReplicaSets(builder.Definition.Namespace).Get(
+		context.TODO(), builder.Definition.Name, metav1.GetOptions{})
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Create builds replicaset in the cluster and stores the created object in struct.
+func (builder *Builder) Create() (*Builder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Creating replicaset %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+	if !builder.Exists() {
+		builder.Object, err = builder.apiClient.ReplicaSets(builder.Definition.Namespace).Create(
+			context.TODO(), builder.Definition, metav1.CreateOptions{})
+		if err == nil {
```

Comment:
Please remove this condition. As you can see on previos line you already set: **builder.Object, err =**